### PR TITLE
Search

### PR DESCRIPTION
**Pull request by Casanova569**
Summary:
The trello card is considered approved when the user can search and filter movies.

Trello Card:
Name: Búsqueda y Filtrado
Link: https://trello.com/c/NGsSwQOo/10-b%C3%BAsqueda-y-filtrado

### CODE REVIEW

File: src/Controllers/MovieAppController.cs [hunk 1]
```diff
@@ -25,9 +29,34 @@ public IUnitOfWork UnitOfWork
         }
 
         [HttpGet]
-        public IActionResult Index()
+        public IActionResult Index(string movieGenre, string searchString)
         {
-            return View(UnitOfWork.Movies.GetAll());
+            var movies = UnitOfWork.Movies.GetAll();
+
```
Comment (truccomarcos) [inline src/Controllers/MovieAppController.cs]: Remove empty line

File: src/Controllers/MovieAppController.cs [hunk 2]
```diff
@@ -25,9 +29,34 @@ public IUnitOfWork UnitOfWork
...
+            if (!string.IsNullOrEmpty(searchString))
+            {
+                movies = movies.Where(s => s.Title.Contains(searchString));
+            }
+
```
Comment (truccomarcos) [inline src/Controllers/MovieAppController.cs]: Same here (empty line)

File: src/Controllers/MovieAppController.cs [hunk 3]
```diff
@@ -25,9 +29,34 @@ public IUnitOfWork UnitOfWork
         }
 
         [HttpGet]
-        public IActionResult Index()
+        public IActionResult Index(string movieGenre, string searchString)
         {
-            return View(UnitOfWork.Movies.GetAll());
+            var movies = UnitOfWork.Movies.GetAll();
+
+            if (!string.IsNullOrEmpty(searchString))
```
Comment (truccomarcos) [inline src/Controllers/MovieAppController.cs:36]: When we have a one line `if` block. we usually write them without the brackets.
```chsarp
if (!string.IsNullOrEmpty(searchString))
    movies = movies.Where(s => s.Title.Contains(searchString));
```

File: src/Controllers/MovieAppController.cs [hunk 4]
```diff
@@ -25,9 +29,34 @@ public IUnitOfWork UnitOfWork
...
+            {
+                movies = movies.Where(s => s.Title.Contains(searchString));
+            }
+
+            if (!string.IsNullOrEmpty(movieGenre))
```
Comment (truccomarcos) [inline src/Controllers/MovieAppController.cs:38]: Same here (one line `if` blocvk)

File: src/Controllers/MovieAppController.cs [hunk 5]
```diff
@@ -25,9 +29,34 @@ public IUnitOfWork UnitOfWork
...
+            {
+                movies = movies.Where(x => x.Genre == movieGenre);
+            }
+
```
Comment (truccomarcos) [inline src/Controllers/MovieAppController.cs]: Same here (empty line)

File: src/Controllers/MovieAppController.cs [hunk 6]
```diff
@@ -25,9 +29,34 @@ public IUnitOfWork UnitOfWork
...
+                Movies = movies.ToList()
+            };
+
+            return View(movieGenreVM);
+        }
+
+        [HttpPost]
+        public string Index(string searchString, bool notUsed)
+        {
+            return "From [HttpPost]Index: filter on " + searchString;
```
Comment (truccomarcos) [inline src/Controllers/MovieAppController.cs]: What is this method for?

File: src/Repositories/Interfaces/IMovieRepository.cs [hunk 7]
```diff
@@ -1,9 +1,11 @@
+using System.Collections.Generic;
 using MvcMovie.Models;
 
 namespace MvcMovie.Repositories.Interfaces
 {
     public interface IMovieRepository : IRepository<Movie>
     {
-        Movie GetMovieWithGenre(string genre);
+        IEnumerable<Movie> GetMovieWithGenre(string genre);
```
Comment (truccomarcos) [inline src/Repositories/Interfaces/IMovieRepository.cs]: This methods are not being used, we should probably remove them if they have no references

File: src/Repositories/MovieRepository.cs [hunk 8]
```diff
@@ -11,14 +12,19 @@ public MovieRepository(MvcMovieContext context) : base(context)
         {
         }
 
-        public Movie GetMovieWithGenre(string genre)
+        public IEnumerable<Movie> GetMovieWithGenre(string genre)
```
Comment (truccomarcos) [inline src/Repositories/MovieRepository.cs]: Same here (Remove unused methods).
Also, try to keep al methods on the top and properties at the bottom of the class.

File: src/Views/MovieApp/Index.cshtml [hunk 9]
```diff
@@ -1,33 +1,44 @@
...
+    <p>
+        <select asp-for="MovieGenre" asp-items="Model.Genres">
+            <option value="">All</option>
+        </select>
+
+        Title: <input type="text" asp-for="SearchString" />
+        <input type="submit" value="Filter" />
+    </p>
+</form>
+@if (Model !=null)
```
Comment (truccomarcos) [inline src/Views/MovieApp/Index.cshtml]: You should probably also check if the `Movies` attribute of the view model has any elements (Model.Movies.Any())

**Review by truccomarcos — CHANGES_REQUESTED** [review]

**Review by truccomarcos — CHANGES_REQUESTED** [review]
When I filter by one genre, and try to filter by another one, I get only a dropdown for the first genre. Try filtering by one genre and then filtering by another one to see the error

File: src/Views/MovieApp/Index.cshtml (hunk 9 above)
Comment (Casanova569) [reply src/Views/MovieApp/Index.cshtml]: I fixed the errors in the pr commentAjax.

**Review by truccomarcos — APPROVED** [review]

**Pull request merged**